uitools/toolkitcpp/cpp/Esri/ArcGISRuntime/Toolkit/PopupViewController.cpp

```diff
@@ -43,7 +53,8 @@ namespace Esri::ArcGISRuntime::Toolkit {
  \endlist
  */
 PopupViewController::PopupViewController(QObject* parent):
-  QObject(parent)
+  QObject(parent),
+  m_popupElementsModel(new GenericListModel(&TextPopupElementViewController::staticMetaObject, this))
```

Comment:
The problem I mentioned in the PR description 😞

Reply:
Approach 1: I tried that idea with a new baseclass that all future PopupElementViewControllers would share which would just contain a PopupElement and the ability to check the popupElementType since that will be shared across all Controllers but no luck initially.

Approach 2: Also did a quick test subclassing QAbstractListModel and making it a little more specific to this use case by containing a QList<QVariant>. I got the TextPopupElements to display but this was quick and rough and there are bugs to work out.

I do feel like Approach 1 should work since it holds a QList<QObject*> which is what I am populating it with but getting that underlying controller object back to be set on the View just isn't working so far. I personally would like to go the first route because it's already there and seems like it should work.

Reply:
Tough one. Either approach sounds viable but approach 1 sounds better if you can make it work.